app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,11 +381,11 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
+      environments.map do |env|
```

Comment:
This behavior isn't the same as before. With your changes you are actually destroying environments; whereas previously the code was just removing them from a collection and returning that collection.